feat: Update Helpers.java

priyanshul-govil opened the pull request:
Details in individual commits.

priyanshul-govil committed:
feat: Added mixColumns()

The MixColumns step of AES encryption creates diffusion in the
state matrix. For this implementation, lookup tables from
Constants.java have been used.
More details at https://en.wikipedia.org/wiki/Rijndael_MixColumns

priyanshul-govil commented:
Need code review on each commit. When all commits are made for Helpers.java, I'll merge on my own.

priyanshul-govil committed:
Merge remote-tracking branch 'upstream/main' into main

viveknathani reviewed (commented):
looks good to me!

viveknathani reviewed (approved):
approved

priyanshul-govil committed:
feat: Added addRoundKey()

addRoundKey() masks the State matrix at every round by XORing every
byte of state with corresponding byte in the round key. The round key
is generated by keyExpansion() which will be added later.

viveknathani reviewed (approved):
perfect

priyanshul-govil committed:
feat: Added keyExpansion()

AES-256 encryption requires fifteen 128-bit round keys. These are
generated using the original 256-bit key.
Constants, variable/method names inside keyExpansion() have been taken
directly from the algorithm on Wikipedia.
Details at: https://en.wikipedia.org/wiki/AES_key_schedule

priyanshul-govil committed:
chore: rename term 'function' -> 'method'

In an Object-Oriented language, the functions are called as methods.
Updated the nomenclature in the comments.

priyanshul-govil committed:
chore: Code was repeating a function body

A function was already made for that purpose. This code was replaced
with the function call to make it DRY.

viveknathani commented on src/secur3dit/crypto/Helpers.java:
```diff
@@ -92,4 +92,129 @@ public static void addRoundKey(char[] state, final char[] key) {
+         * This inner class contains helper functions for keyExpansion.
+         * Details at: https://en.wikipedia.org/wiki/AES_key_schedule#The_key_schedule
+         */
+        final class keyExpansionHelpers {
```
class names should be `UpperCamelCase`

priyanshul-govil replied:
done

priyanshul-govil committed:
chore: Changed class name

Class names should be UpperCamelCase.
Changed keyExpansionHelpers -> KeyExpansionHelpers.

priyanshul-govil commented:
Self-merging this PR.
